### DISCUSSION TITLE
How to destroy Swiper using the WebComponent version?

### DISCUSSION THREAD

**Discussion by erkage** [nolimits4web/swiper#7381] [Q&A]
Hey,

I'm using the WebComponent version for a Vue project and like to handle window resizing to enable Swiper just on mobile.
When I try to destory it with `swiperRef.value.destroy()` I'm getting `swiperRef.value.destroy is not a function`.

If there's no destroy how I can achieve that if the window size is bigger than a number it seems like Swiper never got initialized? And I mean not just on load but dynamically on resize.

I initialize Swiper with this:
```
register();
Object.assign(swiperRef.value, swiperConfig);
swiperRef.value.initialize();
```

Thanks!

**Comment by millertchris** [comment]
@erkage this works for me.

`swiperElRef.swiper.destroy();`

or in your case, the following should work...

`swiperRef.swiper.destroy();`

**Comment by Gusis93** [comment]
Just to jump on this issue, I have the same use case and was able to destroy it. Though when I try to recreate it doesn't work anymore. I try:


`swiperInstance.value?.init(swiperRef.value);`

or simply

`swiperInstance.value?.init();`

And I get this error:
`TypeError: swiper.params is undefined`

Any ideas??
